### PR TITLE
OFI: add error cq to cntr_ep

### PR DESCRIPTION
**Pull request by kseager**
+OFI requirement
+additionally using cq for cntr error handling

Signed-off-by: kseager <kayla.seager@intel.com>

@sayantansur : we've already discussed, but if you could review and merge in, thanks! 

@hppritcha this looks like its fixes the unit test failures we were seeing

### CODE REVIEW

File: src/transport_ofi.c [hunk 1]
```diff
@@ -348,6 +348,14 @@ static inline int bind_resources_to_and_enable_ep(void)
 	return ret;
     }
 
+    /* attach CQ for error handling on cntr EP */
+    ret = fi_ep_bind(shmem_transport_ofi_cntr_epfd,
```
Comment (sayantansur) [inline src/transport_ofi.c]: Super tiny nit. Is this still OK to be called cntr_epfd :-)

**Comment by jdinan** [comment]
👍

**Comment by jdinan** [comment]
The other way to solve this is to check the error count on event counter.  This is what's done in the Portals transport, but it looks like we missed it in the OFI implementation of quiet.  Given that we have the CQ anyway, it's always nice to print an error message (all we know from the counter is that an error occurred, not what caused it).  You might want to add a comment about the event counter approach, for future performance tuning in case binding the the CQ to the EP has a performance impact for some provider.

**Comment by sayantansur** [comment]
@jdinan is right. Should we be calling cq_readerr() or cntr_readerr()? I suppose that's a question for Sean.

@kseager can you follow up with Sean to see what he thinks the right behavior should be?

**Pull request merged**